indy_node/server/domain_req_handler.py

```diff
@@ -210,14 +213,52 @@ def _validateAttrib(self, req: Request):
                                        'attribute for it'.
                                        format(TARGET_NYM))
 
-        if op.get(TARGET_NYM) and op[TARGET_NYM] != req.identifier and \
-                not self.idrCache.getOwnerFor(op[TARGET_NYM],
-                                              isCommitted=False) == origin:
-            raise UnauthorizedClientRequest(
-                req.identifier,
-                req.reqId,
-                "Only identity owner/guardian can add attribute "
-                "for that identity")
+        is_owner = self.idrCache.getOwnerFor(op[TARGET_NYM],
+                                             isCommitted=False) == origin
+        field = None
+        value = None
+        for key in (RAW, ENC, HASH):
+            if key in op:
+                field = key
+                value = op[key]
+                break
+        if field is None or value is None:
+            raise LogicError('Attribute data cannot be empty')
+
+        get_key = None
+        if field == RAW:
+            try:
+                get_key = json.loads(value)
+                if len(get_key) == 0:
+                    raise InvalidClientRequest(origin, req.reqId,
+                                               '"row" attribute field must contain non-empty dict'.
+                                               format(TARGET_NYM))
+                get_key = next(iter(get_key.keys()))
+            except JSONDecodeError:
+                raise InvalidClientRequest(origin, req.reqId,
+                                           'Attribute field must be dict while adding it as a row field'.
+                                           format(TARGET_NYM))
+        else:
+            get_key = value
+
+        if get_key is None:
+            raise LogicError('Attribute data must be parsed')
+
+        old_value, seq_no, _, _ = self.getAttr(op[TARGET_NYM], get_key, field)
```

Comment:
I think we need to call `getAttr` with `isCommitted=False`

Reply:
ok